pkg/api/api_implementation.go

```diff
@@ -79,8 +88,17 @@ func NewAPI(opts ...NewAPIOption) (API, error) {
 // Get the identified object from the engine.
 func (a defaultAPI) Get(ctx context.Context, o types.IdentifiedObject, opts ...types.GetOption) error {
 	options := types.GetOptions{}
+	var err error
+	for _, requestOpt := range a.requestOptions {
+		if getOpt, ok := requestOpt.(types.GetOption); ok {
+			err = errors.Join(err, getOpt.ApplyToGet(&options))
+		}
+	}
 	for _, opt := range opts {
-		opt.ApplyToGet(&options)
+		err = errors.Join(err, opt.ApplyToGet(&options))
+	}
```

Comment:
this can be prettied up a bit for all the request methods with a small generic helper:

```go
func resolveRequestOptions[T any](commonOptions []types.Option, requestOptions []T) []T {
	return append(filterOptions[T](commonOptions), requestOptions...)
}

func filterOptions[T any](opts []types.Option) []T {
	ret := make([]T, 0, len(opts))
	for _, v := range opts {
		if v, ok := v.(T); ok {
			ret = append(ret, v)
		}
	}

	return ret
}
```

With that present, the marked code would be replaced with this:
```go
for _, opt := range resolveRequestOptions(a.requestOptions, opts) {
	err = errors.Join(err, opt.ApplyToGet(&options))
}
```